unittests/CMakeLists.txt

```diff
@@ -87,7 +87,7 @@ macro (create_tests_with_backend NVQIR_BACKEND EXTRA_BACKEND_TESTER)
     set(TEST_LABELS "gpu_required")
   endif()
   if (${NVQIR_BACKEND} STREQUAL "custatevec-fp32")
-    target_compile_definitions(${TEST_EXE_NAME} PRIVATE -DCUDAQ_BACKEND_CUSTATEVEC_FP32)
+    target_compile_definitions(${TEST_EXE_NAME} PRIVATE -DCUDAQ_BACKEND_CUSTATEVEC_FP32 -DCUDAQ_SIMULATION_SCALAR_FP32)
```

Comment:
Q: is there a rationale for having both an `NVQIR_SIMULATION_BACKEND` and an `NVQIR_BACKEND`? They seem sort of redundant at first glance.

Reply:
No rationale that I know of. My guess is two different people implemented that part of `nvq++.in` and `unittests/CMakeLists.txt` and didn't touch base with each other.